buffer/src/test/java/io/netty/buffer/PooledByteBufAllocatorTest.java

```diff
@@ -821,4 +823,64 @@ private void testUsedHeapMemory(int initialCapacity) {
         trimCaches(allocator);
         assertEquals(0, allocator.pinnedDirectMemory());
     }
+
+    @Test
+    public void pinnedMemoryMustReflectBuffersInUse() {
+        PooledByteBufAllocator alloc = newAllocator(true);
```

Comment:
I wonder if we want to run the test one time with cache enabled and one time without.

Reply:
Added a second test.